index.js

```diff
@@ -32,7 +30,87 @@ io.use(socketioJwt.authorize({
 }));
 
 io.on('connection', function(socket) {
-  console.log('A user connected.', socket.decoded_token.name);
+  console.log('user connected');
+
+  socket.on('get user', function(username) {
+    User.findOne({username: username}, (err, user) => {
+      if (err) return socket.emit('get user', err, null);
+      if (!user) {
+        return socket.emit('get user', {message: 'No such user'}, null);
+      }
+
+      socket.emit('get user', err, {
+        username: user.username,
+        rooms: user.rooms
+      });
+    });
+  });
+
+  socket.on('create user', req => {
+    var user = new User({
+      username: req.username,
+      rooms: req.rooms || []
+    });
+
+    user.save()
+    .then(data => {
+      socket.emit('create user', null, 'User created');
+    })
+    .catch(err => {
+      socket.emit('create user', err, null);
+    });
+  });
+
+  socket.on('join room', function(data) {
+    if (!data.user) {
+      return socket.emit('join room', 'Join room requires user', null);
+    }
+    if (!data.room) {
+      return socket.emit('join room', 'no specified room to join', null);
+    }
+
+    User.update(
+      {username: data.user},
+      {$addToSet: {rooms: data.room}},
+      (err, data) => {
+        if (err) return socket.emit('join room', err, null);
+      }
+    );
+
+    socket.join(data.room, function(err) {
+      socket.emit('join room', err, `Joined ${data.room}`);
+    });
+  });
+
+  socket.on('leave room', function(data) {
+    console.log('leaving room');
```

Comment:
Check out the [debug](https://www.npmjs.com/package/debug) module for this system event logging.